# Conditional probability

Some events are dependent on each other. Like drawing out cards out of a deck. Getting an ace in the first draw would certainly affect the probability of getting an ace in the next draw. 

This is called conditional probability. Probability of the event A after event B has happened is:


![](https://i.imgur.com/PfAXRnX.png)

Further, P(A|B) = P(A) * P(B|A)

Let's have a simple card deck with 10 cards numbered sequentially. Cards numbered 0 to 4 (both included) are red and 5 to 9 (both included) are black.
Let us assume we have picked a black card and now we want to find the probability of picking another black card then, by conditional probability:

```
P(two blacks) = P(black) * P(black given the first card was black)
              = 5/10 * (5-1)/(10-1)
              = 0.5 * 0.44
              = 0.22
```

Let's test the above mathematical result empirically by simulating 1 million observations

In [1]:
import numpy as np

class Deck():
    def __init__(self, n=10):
        self.current_deck = [card for card in range(n)]
                
    def random_pick(self):
        chosen_card = np.random.choice(self.current_deck)
        self.current_deck.remove(chosen_card)
        color = 'black'
        if chosen_card < 5:
            color = 'red'
        return chosen_card, color    

In [2]:
N = int(1e6)
blacks = 0
for observation in range(N):
    deck = Deck()
    n1, card1 = deck.random_pick()
    n2, card2 = deck.random_pick()
    if card1 == 'black' and card2 == 'black':
        blacks += 1
P = blacks/N        
print(f"Conditional probablity is: {P}")        

Conditional probablity is: 0.222156


# Conclusion
The mathematically calculated probability was 0.22 and empirically it turns out to be 0.222156 which is pretty close
